client/landing/stepper/declarative-flow/internals/steps-repository/create-site/index.tsx

```diff
@@ -146,11 +147,12 @@ const CreateSite: Step = function CreateSite( { navigation, flow, data } ) {
 	const skipMigration = urlQueryParams.get( 'skipMigration' ) || '';
 	const useThemeHeadstart = ! isStartWritingFlow( flow ) && ! isNewHostedSiteCreationFlow( flow );
 	const shouldGoToCheckout = Boolean( planCartItem );
+	const { select } = useRegistry();
+	const { getSite } = select( SITE_STORE );
 
 	async function createSite() {
-		if ( isManageSiteFlow ) {
-			const slug = getSignupCompleteSlug();
-
+		const slug = getSignupCompleteSlug();
+		if ( isManageSiteFlow || ( slug && getSite?.( slug )?.plan?.is_free ) ) {
```

Comment:
@alshakero referred an if( slug ) check may be enought, but given that this is part of the checkout logic I'm targeting more to my case to decrease the odds of breakage.
This handles the case of going to plans after pressing back. In that case a site already exists, we were handling the case of the user passing by checkout but we should also check for sites on the free plan as in that case a site may exist without the checkout.